smartlearner/interfaces/batch_scheduler.py

```diff
@@ -25,3 +22,11 @@ def givens(self):
     @abstractmethod
     def __iter__(self):
         raise NotImplementedError("Subclass of 'BatchScheduler' must implement '__iter__()'.")
+
+    @abstractmethod
+    def save(self, savedir="./"):
+        raise NotImplementedError("Subclass of 'BatchScheduler' must implement 'save(savedir)'.")
```

Comment:
I'd leave those optionnal. I get that ideally this is what should be done, but I'm not sure we should enforce it.